### ISSUE TITLE
How to handle (non-Python) binary executables to be installed to $PREFIX/bin

### ISSUE THREAD

**Issue by aldanor** [scikit-build/scikit-build#271]
I was wondering how to handle additional binaries generated by CMake (for example, some complicated non-Python CLIs) so that they are installed to `$PREFIX/bin` and then tracked by setuptools?

Getting the first part working is pretty easy, as you can just point CMake at the right prefix and its install command will copy the binaries there. However, setuptools wouldn't be aware of them and as such they won't get properly uninstalled.

**Comment by winni2k** [comment]
I am trying to do the same thing in order to distribute a C++ app through pypi so that it is compiled from source on installation. I think the `scripts` argument to `setup` in the `setup.py` should work: https://python-packaging.readthedocs.io/en/latest/command-line-scripts.html#the-scripts-keyword-argument

However, I can't figure out how to set the location of the "script" to the build directory.

**Comment by jcfr** [comment]
Thanks for the additional feedback.

Would you have a simple example allowing to reproduce the "problem" ?

**Comment by winni2k** [comment]
Actually, I think I was mistaken. The CMake target (a binary called Bifrost) appears to be captured just fine. 

Here is my setup.py:
```
from skbuild import setup 

setup(name='bifrost-src',
      description='',
      version='1.0.4',
      data_files=[
          ('.', ['setup.py'])
      ],
      )
```

**Comment by winni2k** [comment]
Here is the commit of the project I am working on, and it appears to work https://github.com/winni2k/bifrost/commit/bd99b51957c9a361388e542af6558a03ba91efb1

**Comment by winni2k** [comment]
The pip uninstall appears to work for me as well.

**Comment by winni2k** [comment]
I'm pretty sure this use case is one of the reasons conda exists. Any reason why that won't work for you?https://docs.conda.io/en/latest/index.html

**Comment by henryiii** [comment]
See [cmake-python-distributions](https://github.com/scikit-build/cmake-python-distributions) or [clang-format-wheel](https://github.com/ssciwr/clang-format-wheel) for examples of projects doing this.

**Comment by bilke** [comment]
@henryiii Yes I did it like `cmake-python-distributions`. Thanks for the hints.

**Comment by wojdyr** [comment]
> See [cmake-python-distributions](https://github.com/scikit-build/cmake-python-distributions) or [clang-format-wheel](https://github.com/ssciwr/clang-format-wheel) for examples of projects doing this.

It works, but the binary executable is not in $PREFIX/bin, as per the issue title. I see that cmake wheel puts into PATH a python script that imports a python module that uses subprocess to call actual cmake.

Is there a way to put the binary directly in $PREFIX/bin? That's what ruff wheel does, for example. But it is packaged with Rust-specific [build-backend = "maturin"][1].

[1]: https://github.com/charliermarsh/ruff/blob/673aa6e90f96d512f8d9b4c2bb599eb7b2516b27/pyproject.toml

**Comment by henryiii** [comment]
You can use `scikit-build-core`, and put the binary in `SKBUILD_SCRIPTS_DIR`. That also natively supports (via a configuration option) CPython API-free wheels.

**Comment by henryiii** [comment]
scikit-build (classic) after #1185 is now a thin wrapper around scikit-build-core's setuptools plugin. Closing all open issues; if this still applies, please reopen or open an issue on scikit-build-core. Thanks!
